cl_dll/tri.cpp

```diff
@@ -57,4 +162,14 @@ void CL_DLLEXPORT HUD_DrawTransparentTriangles( void )
 
 	if ( g_pParticleMan )
 		 g_pParticleMan->Update();
+
+	if (gHUD.white_sprite == 0)
+		return;
+
+	if (gEngfuncs.pTriAPI->SpriteTexture(const_cast<model_s*>(gEngfuncs.GetSpritePointer(gHUD.white_sprite)), 0))
```

Comment:
`if (!...) return;`?

Reply:
well I actually meant the line that I commented on

Reply:
Honestly, I really don't like the concept of guard clauses when you have not much cases to check validity for on average

I usually prefer for code design to make them into nested if statements than exit from function each time when only x or y is not valid, because it more versality when for example you want later to add some code that should be executed exactly after that

Well, of course it still depends on case by case, but on average I would to prefer nested if statements instead

```cpp
// Pre

if (gHUD.white_sprite)
{
	// Might be there would be here some new code too, who knows?

	if (gEngfuncs.pTriAPI->SpriteTexture(const_cast<model_s*>(gEngfuncs.GetSpritePointer(gHUD.white_sprite)), 0))
	{
		// Triggers code
	}

	// And here too?
}

// Post
```

Reply:
Please make them both `if() return;`, that's what I was asking here. 🙃

> I usually prefer for code design to make them into nested if statements than exit from function each time when only x or y is not valid, because it more versatility when for example you want later to add some code that should be executed exactly after that

Then you turn them nested when you need it. Early return is better because it doesn't indent the code to the right, and because it's easier to follow the control flow. You don't need to keep track of the current conditions in the mental stack, because you consider them once, then they return, and you no longer need to think about them.

Designing for "something we might need to extend in the future" frequently leads to needlessly convoluted code, because right now you don't know what exactly you will need in the future. As a general principle, you should write code for the current needs only, and only in specific rare occasions design with extensibility from the get-go, while already having several use-cases in mind that you can design against.

Reply:
> Please make them both `if() return;`, that's what I was asking here. 🙃

I realized this even before I made the last commit, it’s just my preferences followed me to use nested if statements since there are not many cases that need to be checked for

But so to speak, your argument also makes sense that it is better to expand when you know it needed, like it good to make nested when you don't want to add some new code to it, but at the same time you have roadmap what in future you would exactly add here